src/metabase/lib/drill_thru/sort.cljc

```diff
@@ -46,8 +61,13 @@
   ([query                        :- ::lib.schema/query
     stage-number                 :- :int
     {:keys [column], :as _drill} :- ::lib.schema.drill-thru/drill-thru.sort
-    direction                    :- [:enum :asc :desc]]
-   (lib.order-by/order-by query stage-number column (keyword direction))))
+    direction                    :- ::lib.schema.order-by/direction]
+   ;; if you have an existing order by, the drill thru returned by [[sort-drill]] would only be one that would suggest
+   ;; changing it to the opposite direction, so we can safely assume we want to change the direction and
+   ;; use [[lib.order-by/change-direction]] here.
```

Comment:
I still find it counterintuitive to set `:asc` if the drill says `:sort-directions [:desc]` or if `direction` is `:desc`. Would it be difficult or strange to set `direction`?

Reply:
Yeah, I agree with you and I was going to push a change for this but I guess it auto-merged first. Sorry, I'll fix it in a follow on PR